http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -297,8 +301,8 @@ private URI endpoint() {
         return context.executeRequest().request().getUri();
     }
 
-    private void handleFailure(Supplier<String> msg, Throwable cause) {
-        log.debug(msg.get(), cause);
+    private void handleFailure(Channel channel, Supplier<String> msgSupplier, Throwable cause) {
+        ChannelLogHelper.debug(log, channel, msgSupplier, cause);
```

Comment:
Are there any mechanisms we can consider to help guarantee that `ChannelLogHelper` is used? E.g., a Checkstyle rule in Netty-specific classes that forbids regular logger calls.

Reply:
Yeah I thought about that, but a module-wide checkstyle rule  wouldn't work since there are lots of logging situations that don't happen in the context of a channel: e.g. failure to allocate a channel. I think if we create a more complete logging facade and allow the `channel` parameter to be `null` then that would work, in exchange for some slight clunkiness.

Reply:
Makes sense. I suppose if we take the object-based approach, we could forbid directly instantiating a normal logger (via Checkstyle), and then add more explicit methods that do not require a channel. Users could still bypass channel ID logging, but it would be a much more explicit decision.